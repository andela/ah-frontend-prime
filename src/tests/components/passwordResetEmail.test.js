import React from "react";
import { shallow } from "enzyme";
import PasswordResetEmailComponent from "../../components/passwordResetEmail";

let wrapper;
let props = {
  email: "",
  handleOnChange: jest.fn(),
  errors: {
    email: ""
  },
  handleOnSubmit: jest.fn()
};
describe("passwordResetEmailLink ", () => {
  beforeEach(() => {
    wrapper = shallow(<PasswordResetEmailComponent {...props} />);
  });
  it("should render without fail", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should test email errors field", () => {
    wrapper.setProps({ errors: { email: "anyatibrian@gmail" } });
    expect(wrapper).toMatchSnapshot();
  });
  it("should test the loader field", () => {
    wrapper.setProps({ isLoading: "<Loader/>" });
    expect(wrapper).toMatchSnapshot();
  });
});
