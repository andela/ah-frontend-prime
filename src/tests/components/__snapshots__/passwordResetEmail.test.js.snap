// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwordResetEmailLink  should render without fail 1`] = `ShallowWrapper {}`;

exports[`passwordResetEmailLink  should test email errors field 1`] = `ShallowWrapper {}`;

exports[`passwordResetEmailLink  should test email errors field 2`] = `ShallowWrapper {}`;
